sdk/core/allocator/main.cc

```diff
@@ -782,6 +797,13 @@ namespace
 		  *capability, *chunk, bodySize, isPrecise, reallyFree);
 	}
 
+	bool timeout_is_valid(Timeout *timeout)
+	{
+		return !heap_address_is_valid(timeout) &&
```

Comment:
This is worthy of a comment, at least and should be documented on the APIs that take a `Timeout`. By the way I observe that `heap_address_is_valid` returns true for pointers into the hazardQuarantine but I think that is fine as they should never escape the allocator.

Reply:
Agreed.  I think that should go in timeout.h in the generic coverage of timeouts.  In a follow-on, I am also going to add a check-timeout helper like this, but I want to see with respect to code size whether it makes more sense in the helpers library or as a static inline.

Reply:
Comment added in timeout.h